### DISCUSSION TITLE
Secrets files not working with DUO_API

### DISCUSSION THREAD

**Discussion by jubepue** [authelia/authelia#8480] [Q&A] [answered]
### Version

v4.38.17

### Deployment Method

Docker

### Reverse Proxy

Traefik

### Reverse Proxy Version

3.2.3

### Description

When using variables AUTHELIA_DOU_API_INTEGRATION_KEY_FILE and AUTHELIA_DOU_API_SECRET_KEY_FILE gives an error.

### Reproduction

services:
  authelia:
    image: 'authelia/authelia:latest'
    container_name: 'authelia'
    restart: 'unless-stopped'
    expose:
      - "9091"
    secrets:
      - 'duo_key'
      - 'duo_secret'
      - 'jwt_secret'
      - 'ldap_secret'
      - 'session_secret'
      - 'redis_secret'
      - 'encryption_key'
      - 'db_secret'
      - 'smtp_secret'
      - 'hmac_secret'
    volumes:
      - '/etc/localtime:/etc/localtime:ro'
      - '/etc/docker/volumes/authelia:/config'
      - '/etc/docker/volumes/certs/jubepue.net/ca.pem:/certificates/ca.pem:ro'
    environment:
      PUID: '33’
      PGID: '33'
      TZ: 'Europe/Madrid'
      AUTHELIA_DOU_API_INTEGRATION_KEY_FILE: '/run/secrets/dou_key'
      AUTHELIA_DOU_API_SECRET_KEY_FILE: '/run/secrets/dou_secret'
      AUTHELIA_IDENTITY_VALIDATION_RESET_PASSWORD_JWT_SECRET_FILE: '/run/secrets/jwt_secret'
      AUTHELIA_AUTHENTICATION_BACKEND_LDAP_PASSWORD_FILE: '/run/secrets/ldap_secret'
      AUTHELIA_SESSION_SECRET_FILE: '/run/secrets/session_secret'
      AUTHELIA_SESSION_REDIS_PASSWORD_FILE: '/run/secrets/redis_secret'
      AUTHELIA_STORAGE_ENCRYPTION_KEY_FILE: '/run/secrets/encryption_key'
      AUTHELIA_STORAGE_POSTGRES_PASSWORD_FILE: '/run/secrets/db_secret'
      AUTHELIA_NOTIFIER_SMTP_PASSWORD_FILE: '/run/secrets/smtp_secret'
      AUTHELIA_IDENTITY_PROVIDERS_OIDC_HMAC_SECRET_FILE: '/run/secrets/hmac_secret'
    labels:
      diun.enable: true
      traefik.enable: true
      traefik.http.routers.authelia.rule: 'Host(`auth.example.net`)'
      traefik.http.middlewares.authelia.forwardAuth.address: 'http://authelia:9091/api/authz/forward-huta?authelia_url=https%3A%2F%2Fauth. example.net%2F'
      traefik.http.middlewares.authelia.forwardAuth.trustForwardHeader: true
      traefik.http.middlewares.authelia.forwardAuth.authResponseHeaders: 'Remote-User,Remote-Groups,Remote-Email,Remote-Name'
      traefik.http.middlewares.authelia_basic.forwardAuth.address: 'http://authelia:9091/api/verify?auth=basic'
      traefik.http.middlewares.authelia_basic.forwardAuth.trustForwardHeader: true
      traefik.http.middlewares.authelia_basic.forwardAuth.authResponseHeaders: 'Remote-User,Remote-Groups,Remote-Email,Remote-Name'
      traefik.http.services.authelia.loadbalancer.server.port: '9091'
    networks:
      internal:
      proxy:

secrets:
  duo_key:
    file: '/etc/docker/volumes/secrets/duo_key.txt'
  duo_secret:
    file: '/etc/docker/volumes/secrets/duo_secret.txt'
  jwt_secret:
    file: '/etc/docker/volumes/secrets/jwt_secret.txt'
  ldap_secret:
    file: '/etc/docker/volumes/secrets/ldap_secret.txt'
  session_secret:
    file: '/etc/docker/volumes/secrets/session_secret.txt'
  redis_secret:
    file: '/etc/docker/volumes/secrets/redis_secret.txt'
  encryption_key:
    file: '/etc/docker/volumes/secrets/encryption_key.txt'
  db_secret:
    file: '/etc/docker/volumes/secrets/postgresql_secret.txt'
  smtp_secret:
    file: '/etc/docker/volumes/secrets/smtp_secret.txt'
  hmac_secret:
    file: '/etc/docker/volumes/secrets/hmac_secret.txt'
    
networks:
  internal:
    name: 'backend'
    external: true
  proxy:
    name: 'frontend'
    external: true

### Configuration (Authelia)

```yaml
duo_api:
  disable: false
  hostname: 'api-xxxxxxxxxxx.duosecurity.com'
  # integration_key: ''
  ## Secret can also be set using a secret: https://www.authelia.com/c/secrets
  # secret_key: ''
  enable_self_enrollment: false
```

### Build Information

```shell
Last Tag: v4.38.17
State: tagged clean
Branch: v4.38.17
Commit: 592ab519b4167dfe4462ee0a613d38d3e4b8efa0
Build Number: 35061
Build OS: linux
Build Arch: amd64
Build Compiler: gc
Build Date: Thu, 31 Oct 2024 09:37:31 +1100
Extra: 

Go:
    Version: go1.23.2
    Module Path: github.com/authelia/authelia/v4
    Executable Path: github.com/authelia/authelia/v4/cmd/authelia
```

### Logs (Authelia)

```shell
time="2024-12-19T13:02:03+01:00" level=warning msg="Configuration: configuration environment variable not expected: AUTHELIA_DOU_API_INTEGRATION_KEY_FILE" caller="github.com/authelia/authelia/v4/internal/commands/context.go:317 (*CmdCtx).ConfigValidateLogRunE"
time="2024-12-19T13:02:03+01:00" level=warning msg="Configuration: configuration environment variable not expected: AUTHELIA_DOU_API_SECRET_KEY_FILE" caller="github.com/authelia/authelia/v4/internal/commands/context.go:317 (*CmdCtx).ConfigValidateLogRunE"
time="2024-12-19T13:02:03+01:00" level=error msg="Configuration: duo_api: option 'integration_key' is required when duo is enabled but it's absent" caller="github.com/authelia/authelia/v4/internal/commands/context.go:324 (*CmdCtx).ConfigValidateLogRunE" stack="github.com/authelia/authelia/v4/internal/commands/context.go:324 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:186 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.1/command.go:970                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.1/command.go:1117                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.1/command.go:1041                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\ninternal/runtime/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1700                                         goexit"
time="2024-12-19T13:02:03+01:00" level=error msg="Configuration: duo_api: option 'secret_key' is required when duo is enabled but it's absent" caller="github.com/authelia/authelia/v4/internal/commands/context.go:324 (*CmdCtx).ConfigValidateLogRunE" stack="github.com/authelia/authelia/v4/internal/commands/context.go:324 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:186 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.1/command.go:970                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.1/command.go:1117                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.1/command.go:1041                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\ninternal/runtime/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1700                                         goexit"
time="2024-12-19T13:02:03+01:00" level=fatal msg="Can't continue due to the errors loading the configuration" caller="github.com/authelia/authelia/v4/internal/commands/context.go:327 (*CmdCtx).ConfigValidateLogRunE" stack="github.com/authelia/authelia/v4/internal/commands/context.go:327 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:186 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.1/command.go:970                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.1/command.go:1117                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.1/command.go:1041                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\ninternal/runtime/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1700                                         goexit"
```

### Generative AI

No

### Pre-Submission Checklist

- [X] I agree to follow the [Code of Conduct](http://www.authelia.com/code-of-conduct)

- [X] This is a bug report and not a support request

- [X] I have read the security policy and this bug report is not a security issue or security related issue

- [X] I have either included the complete configuration file or I am sure it's unrelated to the configuration

- [X] I have either included the complete debug / trace logs or the output of the build-info command if the logs are not relevant

- [X] I have provided all of the required information in full with the only alteration being reasonable sanitization in accordance with the [Troubleshooting Sanitization](https://www.authelia.com/r/sanitize) reference guide

- [X] I have checked for related proxy or application logs and included them if available

- [X] I have checked for related issues and checked the documentation

**Comment by nightah** [comment, marked as answer]
You've misspelled `DUO` and instead are using `DOU`.
